example-scene/src/turrets.ts

```diff
@@ -0,0 +1,138 @@
+  MeshRenderer.setBox(cube)
+  return segment
+}
+
+export function initRaycastTurrets(parent: Entity) {
+  const floor = engine.addEntity()
+  Transform.create(floor, {
+    parent,
+    position: { x: 8, y: 0, z: 8 },
+    scale: { x: 16, y: 0.01, z: 16 },
+  })
+  MeshRenderer.setBox(floor)
+  const collisionMask = ColliderLayer.CL_PHYSICS
+  MeshCollider.setBox(floor, collisionMask)
+
+
+  {
+    const parentEntity = engine.addEntity()
+    const parentTransform = Transform.create(parentEntity, { parent })
+
+    parentTransform.position.x = 12
+    parentTransform.position.y = 5
+    parentTransform.position.z = 12
+
+    const segment1 = addTurretSegment(parentEntity, 1)
+    const segment2 = addTurretSegment(segment1, 2)
+    const segment3 = addTurretSegment(segment2, 3)
+    const segment4 = addTurretSegment(segment3, 4)
+    const segment5 = addTurretSegment(segment4, 6)
+
+    Raycast.create(segment5, {
+      originOffset: Vector3.scale(Vector3.Forward(), 1.1),
+      direction: {
+        $case: 'localDirection',
+        localDirection: Vector3.Forward()
+      },
+      continuous: true,
+      maxDistance: 999,
+      queryType: RaycastQueryType.RQT_HIT_FIRST
+    })
+  }
+
+  {
+    const parentEntity = engine.addEntity()
+    const parentTransform = Transform.create(parentEntity, { parent })
+
+    parentTransform.position.x = 8
+    parentTransform.position.y = 8
+    parentTransform.position.z = 8
+
+    const segment1 = addTurretSegment(parentEntity, 1)
+    const segment2 = addTurretSegment(segment1, 9)
+    const segment3 = addTurretSegment(segment2, 3)
+    const segment4 = addTurretSegment(segment3, 4)
+    const segment5 = addTurretSegment(segment4, 4)
+
+    Raycast.create(segment5, {
+      originOffset: Vector3.scale(Vector3.Forward(), 1.1),
+      direction: {
+        $case: 'globalTarget',
+        globalTarget: Vector3.Zero()
+      },
+      continuous: true,
+      maxDistance: 999,
+      queryType: RaycastQueryType.RQT_HIT_FIRST
+    })
+
+    let i = 0
+
+    engine.addSystem(dt => {
+      const cameraTransform = Transform.get(engine.CameraEntity)
+
+      const globalTarget = Vector3.add(cameraTransform.position, Vector3.Down())
+
+      i += dt
+      globalTarget.x += Math.sin(i)
+      globalTarget.z += Math.cos(i)
+
+      Raycast.getMutable(segment5).direction = { $case: 'globalTarget', globalTarget }
+    })
+  }
+
+  {
+    const parentEntity = engine.addEntity()
+    const parentTransform = Transform.create(parentEntity, { parent })
+
+    parentTransform.position.x = 4
+    parentTransform.position.y = 5
+    parentTransform.position.z = 4
+
+    const segment1 = addTurretSegment(parentEntity, 1)
+    const segment2 = addTurretSegment(segment1, 3)
+    const segment3 = addTurretSegment(segment2, 3)
+    const segment4 = addTurretSegment(segment3, 3)
+    const segment5 = addTurretSegment(segment4, 6)
```

Comment:
Second time I see this code, do you think this would benefit from being a `createTurret` function that returns `segment5`? It would DRY it, and avoid filling the local scope with segments.